rustls/src/lib.rs

```diff
@@ -425,14 +420,14 @@ pub mod client {
     };
     pub use handy::ClientSessionMemoryCache;
 
-    #[cfg(feature = "dangerous_configuration")]
-    pub use crate::verify::{HandshakeSignatureValid, ServerCertVerified, ServerCertVerifier};
-    #[cfg(feature = "dangerous_configuration")]
-    pub use crate::webpki::{
-        verify_server_cert_signed_by_trust_anchor, verify_server_name, WebPkiServerVerifier,
-    };
-    #[cfg(feature = "dangerous_configuration")]
-    pub use client_conn::danger::DangerousClientConfig;
+    /// Dangerous configuration that should be audited and used with extreme care.
+    pub mod danger {
+        pub use super::client_conn::danger::DangerousClientConfig;
+        pub use crate::verify::{HandshakeSignatureValid, ServerCertVerified, ServerCertVerifier};
+        pub use crate::webpki::{
+            verify_server_cert_signed_by_trust_anchor, verify_server_name, WebPkiServerVerifier,
```

Comment:
Seems odd that `rustls::server::danger::ParsedCertificate` is required to use the `rustls::client::danger::verify_server_name` API. (This isn't an issue introduced in this PR, just making a note here.)

Reply:
This is presumably because we want to reuse the same `ParsedCertificate` across calls to `verify_server_cert_signed_by_trust_anchor()` and `verify_server_name()`, right? I think that makes sense?

Reply:
I removed `ParsedCertificate` from the `danger` mod.